### PR TITLE
[Container]  Add secrets for container instances

### PR DESCRIPTION
**Pull request by paulmey**
Add secrets for container instances on create/update.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5578
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-container/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.1.19
+++++++
+* Add '--secrets' and '--secrets-mount-path' options to 'az container create' for usgin secrets in ACI
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-container/HISTORY.rst]: typo: usgin

**Comment by paulmey** [comment]
Thanks @derekbekoe, I fixed the typo

**Comment by paulmey** [comment]
Now fixing some tests...

**Comment by paulmey** [comment]
Fixed a type, a pylint issue and re-recorded the test. Please review.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py [hunk 2]
```diff
@@ -27,6 +28,13 @@ def _environment_variables_type(value):
     return {'name': env_name, 'value': env_value}
 
 
+secrets_type = CLIArgumentType(
+    validator=validate_secrets,
+    help="space-separated secrets in 'key=value' format.",
+    nargs='*'
```
Comment (tjprescott) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py]: nargs='*' means it takes 0 or more arguments. Is this what you intended?  1 or more is nargs='+'

Comment (paulmey) [reply src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py]: Good catch, I think that `+` makes more sense. I was 'inspired' by the `--tags` pattern, but that has a valid use case for 'empty' tags array.
Pending...

**Pull request merged**